src/pathfind/teleport.cpp

```diff
@@ -43,19 +45,19 @@ namespace {
 // This constructor is *only* meant for loading from saves
 teleport_group::teleport_group(const config& cfg) : cfg_(cfg), reversed_(cfg["reversed"].to_bool(false)), id_(cfg["id"])
 {
-	assert(cfg.has_attribute("id"));
-	assert(cfg.has_attribute("reversed"));
+	VALIDATE(cfg.has_attribute("id"), missing_mandatory_wml_key("tunnel", "id"));
+	VALIDATE(cfg.has_attribute("reversed"), missing_mandatory_wml_key("tunnel", "reversed"));
 
-	assert(cfg_.child_count("source") == 1);
-	assert(cfg_.child_count("target") == 1);
-	assert(cfg_.child_count("filter") == 1);
+	VALIDATE(cfg_.child_count("source") == 1, _("config should have one 'source' child."));
```

Comment:
Per gfgtdf's comment:
> Also we are a in a string freeze, maybe it'd be better not to makr these strings as translatable, and to mark the translation mark later.

these should be left as untranslatable strings for now.

Reply:
Still an issue with commit 6d2839c